packages/liveblocks-react-ui/src/primitives/Collapsible/index.tsx

```diff
@@ -149,3 +170,16 @@ export const Content = forwardRef<HTMLDivElement, ContentProps>(
     );
   }
 );
+
+if (process.env.NODE_ENV !== "production") {
+  CollapsibleContent.displayName = COLLAPSIBLE_CONTENT_NAME;
+  CollapsibleRoot.displayName = COLLAPSIBLE_ROOT_NAME;
+  CollapsibleTrigger.displayName = COLLAPSIBLE_TRIGGER_NAME;
+}
```

Comment:
@marcbouchenoire Why do we need these? My understanding was that if `displayName` is not set, React infers it from the name of the function that defines the component. In our case, we're okay with the function name being used for debugging purpose, no?

Reply:
I thought it was because of `forwardRef` but at this point we're just moving these from primitive to primitive and I don't even know if it's needed anymore, plus we don't do it for the high-level components even though they're also using `forwardRef`. I'll test this today but I'll be happy to remove all this boilerplate from all primitives 👍

Reply:
It's not exactly because of `forwardRef`, it's because of the anonymous `() => {}` function inside, if I remove the manual `displayName` then it turns into "Anonymous".

If I use a named function like `const AiMessageContent = forwardRef(function AiMessageContent() { ... })`, it does show up but its name is messed with by the bundler 😢

![](https://github.com/user-attachments/assets/89ce1fa1-9d0a-4d5e-8525-59d37305ce96)

Any other ideas?

Reply:
😵‍💫 Wow that is not ideal - thanks for looking into this! Thank god React 19 removes the need to use `forwardRef`